Rename sysroot directories to be more clear (#103286)
rust-lang/rust: src/bootstrap/CHANGELOG.md

## [Changes since the last major version]

> 

Reviewer: ```suggestion

## [Version 3] - 2022-10-20

```

Author: @jyn514 done, but somehow still the build is not going thru.

Reviewer: Did you see the changes I suggested in https://github.com/rust-lang/rust/pull/103286#pullrequestreview-1149659205 ?

Author: @jyn514 seems some issue with extraction of file to bootstrap-sysroot. 

```
warning: there have been changes to x.py since you last updated.
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, update `config.toml` to use `changelog-seen = 3` instead
downloading https://static.rust-lang.org/dist/2022-09-20/rustfmt-nightly-aarch64-apple-darwin.tar.xz
############################################################################################################################################################################################################### 100.0%
extracting /Users/chetankokil/git/rust/build/cache/2022-09-20/rustfmt-nightly-aarch64-apple-darwin.tar.xz to /Users/chetankokil/git/rust/build/aarch64-apple-darwin/bootstrap-sysroot

failed to execute command: "/Users/chetankokil/git/rust/build/aarch64-apple-darwin/bootstrap-sysroot/bin/rustc" "--print" "sysroot"
error: No such file or directory (os error 2)
```.

This is not allowing me to build or push

Author: ```
ls build/aarch64-apple-darwin build/aarch64-apple-darwin/bootstrap-sysroot/bin
ls: build/aarch64-apple-darwin/bootstrap-sysroot/bin: No such file or directory
build/aarch64-apple-darwin:
bootstrap-sysroot
```

Above is the output of `ls build/aarch64-apple-darwin build/aarch64-apple-darwin/bootstrap-sysroot/bin`

> - These are effectively user facing because they're used for `rustup link`. So far this has worked reasonably well, but after #101691 stage0-sysroot will become user facing as well, at which point it will become extremely confusing that there's both that and stage0. [#103286](https://github.com/rust-lang/rust/pull/103286)

Reviewer: ```suggestion
- The names of the sysroot directories have changed. Previously they were `stage0`, `stage0-sysroot`, `stage1`, and `stage2`. Now they are `bootstrap-sysroot`, `stage0-sysroot`, `stage1-sysroot`, and `stage2-sysroot`. If you've created a rustup toolchain using these sysroots with `rustup toolchain link mytoolchain build/stage1`, you will need to update them by running `rustup toolchain remove mytoolchain && rustup toolchain link mytoolchain build/stage1`. [#103286](https://github.com/rust-lang/rust/pull/103286)
```

## [Changes since the last major version]

## [Version 3] - 2022-10-20

> - The names of the sysroot directories have changed. Previously they were `stage0`, `stage0-sysroot`, `stage1`, and `stage2`. Now they are `bootstrap-sysroot`, `stage0-sysroot`, `stage1-sysroot`, and `stage2-sysroot`. If you've created a rustup toolchain using these sysroots with `rustup toolchain link mytoolchain build/stage1`, you will need to update them by running `rustup toolchain remove mytoolchain && rustup toolchain link mytoolchain build/stage1`. [#103286](https://github.com/rust-lang/rust/pull/103286)

Reviewer: ```suggestion
- The names of the sysroot directories have changed. Previously they were `stage0`, `stage0-sysroot`, `stage1`, and `stage2`. Now they are `bootstrap-sysroot`, `stage0-sysroot`, `stage1-sysroot`, and `stage2-sysroot`. If you've created a rustup toolchain using these sysroots with `rustup toolchain link mytoolchain build/stage1`, you will need to update them by running `rustup toolchain remove mytoolchain && rustup toolchain link mytoolchain build/stage1-sysroot`. [#103286](https://github.com/rust-lang/rust/pull/103286)
```
